src/vm/eventreporter.cpp

```diff
@@ -106,7 +108,16 @@ EventReporter::EventReporter(EventReporterType type)
         {
             m_Description.Append(ssMessage);
             m_Description.Append(W("\n"));
-        }        
+        }
+    }
+
+    // Log the .NET Core Version if we can get it
+    LPCWSTR fxProductVersion = Configuration::GetKnobStringValue(W("FX_PRODUCT_VERSION"));
```

Comment:
Have you verified that `Configuration::*` has the `FX_PRODUCT_VERSION` key/value pair set?

We have several different key/value pair sets. I believe that `FX_PRODUCT_VERSION` is in the one that comes from `CorHost2::_CreateAppDomain`, not this one.

Reply:
I checked how ```FX_PRODUCT_VERSION``` will propagate through the runtime from core-setup and this is what I found:
```
    // Initialize CoreCLR
    coreclr::host_handle_t host_handle;
    coreclr::domain_id_t domain_id;
    auto hr = coreclr::initialize(
        host_path.data(),
        "clrhost",
        property_keys.data(),
        property_values.data(),
        property_size,
        &host_handle,
&domain_id);
```
https://github.com/dotnet/core-setup/blob/bfb18700ea2fb0e600a034ac7f719aee2618a87e/src/corehost/cli/hostpolicy.cpp#L263

 
```coreclr_initialize``` will then call ```Configuration::InitializeConfigurationKnobs``` which sets the key/value pair in Configuration::* with ```property_keys``` and ```property_values``` which contains the FX_PRODUCT_VERSION.

Reply:
Ok, I was not aware of this path.